aries_cloudagent/protocols/coordinate_mediation/v1_0/handlers/mediation_denied_handler.py

```diff
@@ -0,0 +1,42 @@
+"""Handler for incoming mediation-deny-request messages."""
+
+from .....messaging.base_handler import (
+    BaseHandler,
+    BaseResponder,
+    HandlerException,
+    RequestContext,
+)
+
+from ..manager import MediationManager
+from ..messages.mediate_grant import MediationGrant
+from ..models.mediation_record import MediationRecord
+from aries_cloudagent.storage.error import StorageNotFoundError
+
+
+class MediationDenyHandler(BaseHandler):
+    """Handler for incoming mediation denied messages."""
+
+    async def handle(self, context: RequestContext, responder: BaseResponder):
+        """Message handler implementation."""
+        self._logger.debug(
+            "%s called with context %s", self.__class__.__name__, context
+        )
+        assert isinstance(context.message, MediationGrant)
+
+        if not context.connection_ready:
+            raise HandlerException("Invalid mediation request: no active connection")
+        try:
+            mgr = MediationManager(context)
+            _record = await MediationRecord.retrieve_by_id(
+                context, context.mediation_id
+            )
+            await mgr.deny_request(
+                mediation=_record,
+                mediator_terms=context.mediator_terms,
+                recipient_terms=context.recipient_terms
+            )
+        except StorageNotFoundError as err:
+            pass
+        else:
+            pass  # if not existing record, do nothing
+            # TODO: ?create if not existing?
```

Comment:
Send problem report or raise handler exception as on line 27.

Reply:
I think the question can be phrased as, "what should aca-py do when it receives a mediation denied message for a connection with no mediation?". Currently, aca-py quietly does nothing, we could send a problem report back on the connection, but rfc-0211 does not define problem reports, raising an error does not feel correct because nothing has errored. What is the desired behavior?

Reply:
We're already raising an exception on line 27 when the connection isn't in an acceptable state so I'm not sure what the difference is here. The RFC probably should have problem reports defined but problem reports are a "courtesy" and are not strictly speaking required. It's up to the agent developer and the use case to determine how chatty agents are. If you're really against raising the error (which just gets spit out by the dispatcher to the logs anyways), then I would recommend logging something at least.

Reply:
I see your point and agree. I will add the needed report and exception.